deployer/pipeline_deployer.py

```diff
@@ -106,6 +109,21 @@ def _check_experiment_name(self, experiment_name: Optional[str] = None) -> str:
 
         return experiment_name
 
+    def _check_pipeline_display_name(self, tag: Optional[str] = None) -> None:
+        now_str = datetime.now().strftime("%Y%m%d%H%M%S")
+
+        if self.pipeline_display_name is None:
+            self.pipeline_display_name = f"{self.pipeline_name}"
+            logger.info(f"display_name not provided, using {self.pipeline_name} as base")
+        if tag:
+            self.pipeline_display_name += f"-{tag}"
+        self.pipeline_display_name += f"-{now_str}"
+
+        self.pipeline_display_name = self.pipeline_display_name.replace("_", "-")
+        self.pipeline_display_name = self.pipeline_display_name[:127]
```

Comment:
why 127 ?

Reply:
Because the job_id, in vertex AI python SDK, must conform to the following [pattern](https://github.com/googleapis/python-aiplatform/blob/copybara_616969418/google/cloud/aiplatform/constants/pipeline.py): 
```
# Pattern for valid names used as a Vertex resource name.
_VALID_NAME_PATTERN = re.compile("^[a-z][-a-z0-9]{0,127}$", re.IGNORECASE)
```